src/Field/BaseFieldDescription.php

```diff
@@ -0,0 +1,326 @@
+
+    /**
+     * @var array the ORM field information
+     */
+    protected $fieldMapping = [];
+
+    /**
+     * @var array the ORM parent mapping association
+     */
+    protected $parentAssociationMappings = [];
+
+    /**
+     * @var string the template name
+     */
+    protected $template;
+
+    /**
+     * @var array the option collection
+     */
+    protected $options = [];
+
+    /**
+     * @var array[] cached object field getters
+     */
+    private static $fieldGetters = [];
+
+    public function setFieldName(string $fieldName): void
+    {
+        $this->fieldName = $fieldName;
+    }
+
+    public function getFieldName(): ?string
+    {
+        return $this->fieldName;
+    }
+
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+
+        if (!$this->getFieldName()) {
+            $this->setFieldName(substr(strrchr('.'.$name, '.'), 1));
+        }
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setTemplate(string $template): void
+    {
+        $this->template = $template;
+    }
+
+    public function getTemplate(): ?string
+    {
+        return $this->template;
+    }
+
+    /**
+     * @param int|string $type
+     */
+    public function setType($type): void
+    {
+        $this->type = $type;
+    }
+
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * @param mixed $default
+     *
+     * @return mixed
+     */
+    public function getOption(string $name, $default = null)
+    {
+        return isset($this->options[$name]) ? $this->options[$name] : $default;
+    }
+
+    /**
+     * @param mixed $value
+     */
+    public function setOption(string $name, $value): void
+    {
+        $this->options[$name] = $value;
+    }
+
+    public function setOptions(array $options): void
+    {
+        // set the type if provided
+        if (isset($options['type'])) {
+            $this->setType($options['type']);
+            unset($options['type']);
+        }
+
+        // set the template if provided
+        if (isset($options['template'])) {
+            $this->setTemplate($options['template']);
+            unset($options['template']);
+        }
+
+        $this->options = $options;
+    }
+
+    public function getOptions(): array
+    {
+        return $this->options;
+    }
+
+    public function mergeOption(string $name, array $options = []): void
+    {
+        if (!isset($this->options[$name])) {
+            $this->options[$name] = [];
+        }
+
+        if (!\is_array($this->options[$name])) {
+            throw new \RuntimeException(sprintf('The key `%s` does not point to an array value', $name));
+        }
+
+        $this->options[$name] = array_merge($this->options[$name], $options);
+    }
+
+    public function mergeOptions(array $options = []): void
+    {
+        $this->setOptions(array_merge_recursive($this->options, $options));
```

Comment:
Maybe we could try to avoid calling `array_merge_recursive()` (see https://github.com/sonata-project/SonataAdminBundle/pull/6078#issue-412703816).

Reply:
I didn't understood the issue with `array_merge_recursive`. Can you tell me more about it ?

Reply:
My main concern was to create a FieldDescriptionInterface with the same behaviour than the AdminBundle one in order to stop using some duplicated classes and start using the DatagridBundle instead.

This PR https://github.com/sonata-project/SonataAdminBundle/pull/6078 could be done on this bundle when ready I think.

Reply:
>I didn't understood the issue with array_merge_recursive. Can you tell me more about it ?

The "issue" with `array_merge_recursive()` is the result it produces if some key is repeated between their arguments. In this example, it converts a value that should be a boolean into an array:
>The option "virtual_field" with value array is expected to be of type "bool", but is of type "array".

See https://travis-ci.org/github/sonata-project/SonataAdminBundle/jobs/683117311#L758.

Given that situation, I'd like to avoid the propagation of that issue.

Reply:
WDYT about my solution @phansys ? :)

Reply:
>WDYT about my solution @phansys ? :)

I agree.